docs/schulcloud-server/Coding-Guidelines/modules-submodules.md

```diff
@@ -0,0 +1,70 @@
+# Implementation and usage of modules, submodule and barrel files in our project
+
+In this guide, we'll cover how to use modules, submodules and barrel files in our project. These concepts help you organize your code into separate files and directories, making it easier to manage and maintain.
+
+![Module Structure](./../img/Modules-SubModules_background.svg)
+
+## Modules and Submodules
+
+In our project, modules are a way to create separate scopes. This means that interfaces, use cases, services, etc., declared in a module are not visible outside the module unless they are explicitly exported using the `export` keyword. To import a module, you use the `import` keyword followed by the module name. Here's an example:
+
+```typescript
+import { ModuleName } from '@modules/module-name';
+```
+
+Submodules are modules that are part of a larger module. They should be used within the main module. If it is necessary to use parts of the submodule outside the main module, the main module should export this via its barrel file(index.ts):
+
+```typescript
+// @modules/module-name/index.ts
+export { SubmoduleServiceName } from './submodule-name/service.ts';
+```
+
+## Barrel Files
+
+Barrel files are a way to rollup exports from several modules into a single convenient module. The barrel itself is a module file that re-exports selected exports of other modules.
```

Comment:
"module" is overloaded in meaning here, both js modules and nest modules. Maybe that should be made more clear?
The reader is definetly not meant to combine the exports from different nest modules into a more convenient all-encompassing module...

Reply:
maybe use "file" instead of jsmodule, even though module is the technically correct term here.

Reply:
ah, if you mean submodules, please clarify that also :)